holidays/financial/us_new_york.py

```diff
@@ -0,0 +1,77 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from holidays.financial._sifma import SIFMAHolidays
+
+
+class USNewYork(SIFMAHolidays):
+    """US New York (USNY) financial market holidays.
+
+    The USNY calendar represents general "New York business days" used in financial contracts
+    (ISDA derivatives, OTC instruments). While it follows SIFMA (Securities Industry and
+    Financial Markets Association) recommendations for most holidays, it differs from both
+    USGS (US Government Securities) and NYSE calendars.
+
+    Key differences:
+        - vs USGS: USNY does NOT observe Good Friday (USGS does)
+        - vs NYSE: USNY DOES observe Columbus Day and Veterans Day (NYSE does not in modern times)
+        - vs NYSE: USNY does NOT include special one-off closures (NYSE does)
+
+    This calendar includes both full market closures (PUBLIC holidays) and early closes at
+    2:00 PM Eastern Time (HALF_DAY holidays) as recommended by SIFMA, with the exceptions
+    noted above.
+
+    The USNY identifier is used for compatibility with systems that reference New York
+    market holidays specifically, such as OpenGamma Strata and other financial libraries.
+
+    References:
+        * <https://www.sifma.org/resources/general/holiday-schedule/>
+        * <https://web.archive.org/web/20250210040000/https://www.sifma.org/resources/general/holiday-schedule/>
+        * <https://strata.opengamma.io/apidocs/com/opengamma/strata/basics/date/HolidayCalendarIds.html>
+        * <https://github.com/OpenGamma/Strata/blob/main/modules/basics/src/main/java/com/opengamma/strata/basics/date/GlobalHolidayCalendars.java>
+
+    Historical references:
+        * <https://web.archive.org/web/20250210040000/https://www.sifma.org/resources/general/us-holiday-archive/>
+    """
+
+    market = "USNY"
+
+    def _populate_public_holidays(self):
+        # Call parent to get all SIFMA holidays
+        super()._populate_public_holidays()
+
```

Comment:
It is more logical to take a less complex entity as a basis and add elements to it than to take a more complex one and then remove elements. In general, if you do not plan to add any other markets based on SIFMA, it would be optimal to implement the USNY class and, based on it, USGS (adding Good Friday), without a separate SIFMA class.

Reply:
I think it's better to do everything properly from the start.

BTW, about tests: I don't review the tests in detail until the code for the holidays themselves is fully agreed upon, but looking at them now, I see that they don't quite meet our latest standards for tests.